crates/pallet-domains/src/staking.rs

```diff
@@ -52,9 +52,9 @@ impl SharePrice {
             // If there are no shares or no stake, it is valid, but there is no share price.
             Ok(SharePrice(None))
```

Comment:
I am concerned that this edge case might be exploitable. I don't think it is correct to convert shares and stake 1:1 when either original value was zero. This could end up significantly favouring the user.

Instead, the result should be:
- zero stake: if there is no stake, then when converting or rewarding them we should get zero shares, and when attempting to deposit we should get a `MinimumOperatorStake` (or nominator stake) error
- zero shares: if there are no shares, then when converting or rewarding them we should get zero stake, and when attempting to withdraw we should get an `InsufficientShares` error

This means keeping this code, but in the conversion functions, returning `None` (instead of 1:1 conversion), and handling the zeroes/errors in the caller.

Reply:
Answering Ning's question that was posted privately:

It might be possible to end up with a very small remaining stake when we subtract a correctly rounded down stake value. If that happens, the next `share_price` calculation could be zero (or close to zero) rounded down.

But I think this is covered by the `MinimumNominatorStake` (or operator stake). If the small amount is less than the minimum stake, then the entire stake will be withdrawn. If it is greater (rare, only with very large stakes), then the nominator/operator will continue, and can withdraw the rest as normal.

Reply:
> If the small amount is less than the minimum stake, then the entire stake will be withdrawn. If it is greater (rare, only with very large stakes), then the nominator/operator will continue, and can withdraw the rest as normal.

Except we only do that for nominators, not operators?